syntax = "proto3";

option java_multiple_files = true;
option java_package = "io.grpc.examples.iotservice";
option java_outer_classname = "IoTServiceProto";
option objc_class_prefix = "TEMP";

package sensor_service;

service SensorService {
  
  // Registra o dado de um sensor
  rpc RegistrarDadoSensor (EmployeeData) returns (StatusReply) {}
  
  // Consulta os dados de um sensor pela data
  rpc ConsultarDadosSensoresPorData (Data) returns (ListaDadosSensores) {}
  
  // Consulta os dados de um sensor pela localização
  rpc ConsultarDadosSensoresPorLocalizacao (localizacao) returns (ListaDadosSensores) {}
  
}

message DadoSensor {
  uint32 id = 1;
  string data = 2;
  string localizacao = 3;
  float temperatura = 4;
}

message Data {
  string data = 1;
}

message Localizacao {
  string localizacao;
}

message ListaDadosSensores{
  repeated DadoSensor sensor_dado = 1;
}

message EmptyMessage{}


